File: 2015/day_08/part1.c
#include <stdio.h>
#include <string.h>

// solution time: 02:38.22
int main() {
    // read input from file
    FILE *fp = fopen("input.in", "r");

    if (fp == NULL) {
        printf("Could not open file input.in\n");
        return 1;
    }

    char line[256];
    int total_code_chars = 0;
    int total_memory_chars = 0;

    while (fgets(line, sizeof(line), fp)) {
        int line_len = strlen(line) - 1;  // -1 skips \n
        total_code_chars += line_len;

        // 1 to len-1 to exclude the quotes
        for (int i = 1; i < line_len - 1; i++) {
            if (line[i] == '\\') {
                if (line[i + 1] == '\\' || line[i + 1] == '\"') {
                    i++;  // Skip the next character
                } else if (line[i + 1] == 'x') {
                    i += 3;
                }
            }
            total_memory_chars++;
        }
    }

    fclose(fp);

    printf("%d\n", total_code_chars - total_memory_chars);

    return 0;
}

File: 2016/day_06/part1.py
# open and read the file to a string
input = open("input.in", "r")
input = input.read()
input = input.split("\n")

# Solution 1:33
freqs = [{} for i in range(len(input[0]))]

for line in input:
    for i in range(len(line)):
        if line[i] in freqs[i]:
            freqs[i][line[i]] += 1
        else:
            freqs[i][line[i]] = 1

message = ""
for freq in freqs:
    message += max(freq, key=freq.get)
print(message)
